operator/pkg/ingress/ingress.go

```diff
@@ -157,7 +161,7 @@ func NewController(clientset k8sClient.Clientset, options ...Option) (*Controlle
 		nil,
 	)
 
-	ingressClassManager, err := newIngressClassManager(clientset, ic.queue, opts.MaxRetries)
+	ingressClassManager, err := newIngressClassManager(context.Background(), ic.queue, ingressClasses)
```

Comment:
Good idea :+1:

Reply:
To do the plumbing I think we'd need to adjust `NewController` and `Controller.Run` to take a context as a parameter. This might be confusing since only the ingress class manager would actually use the context, and otherwise it would be ignored. Do you think we should also adjust the Ingress Controller and its sub-components to use the context? Or is OK to leave it for now?

Reply:
Yep, this isn't perfect, but I still think it is worth to improve at least the ingress class manager. We can then plumb the context into the other pieces of the controller as soon as we move to the new Resource[T] framework for all the other resources. But I'd leave this for a follow up PR.

Reply:
Sounds like a plan!